modules/controlfreec/1.0/controlfreec.smk

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Jasper
+# Module Author:    Jasper
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+
+# `CFG` is a shortcut to `config["lcr-modules"]["controlfreec"]`
+CFG = op.setup_module(
+    name = "controlfreec",
+    version = "1.0",
+    subdirectories = ["inputs", "run", "outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _controlfreec_input_bam,
+    _controlfreec_config,
+    _controlfreec_run,
+    _controlfreec_calc_sig,
+    _controlfreec_plot,
+    _controlfreec_all,
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _controlfreec_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+
+rule _controlfreec_config:
+    input:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam"
+    output:
+        CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/config_WGS.txt"
+    conda:
+        CFG["conda_envs"]["sambamba"]
+    params:
+        config = CFG["options"]["configFile"],
+        outdir = CFG["dirs"]["run"] + "{seq_type}--{genome_build}/{sample_id}/",
+        window = CFG["options"]["window"],
+        ploidy = CFG["options"]["ploidy"],
+        chrLen = CFG["options"]["chrLenFile"],
+        chrFiles = CFG["options"]["chrFiles"],
+        reference = CFG["options"]["gemMappabilityFile"]
+    shell:
```

Comment:
Are all these tools available in a single conda environment? If not, you need to make a conda environment that has them all available and supply it with this module.

Reply:
Compiled all envs into one